rest/config.go

```diff
@@ -837,3 +851,37 @@ func ServerMain(runMode SyncGatewayRunMode) {
 	ValidateConfigOrPanic(runMode)
 	RunServer(config)
 }
+
+// localhostAddr takes an address like ':4984' or '127.0.0.1:4984' and prefixes with localhost if required.
+func localhostAddr(addr string) string {
+	port := regexp.MustCompile("^:\\d+$")
+	if port.MatchString(addr) {
+		return "localhost" + addr
+	}
+	return addr
+}
+
+// waitForResponse blocks until the given address returns a HTTP 200 response at the root
+func waitForResponse(addr string) error {
+	// TODO: Could this ever be https-only?
+	url := "http://" + localhostAddr(addr) + "/"
+
+	err, _ := base.RetryLoop(
+		"GET "+url,
```

Comment:
I think it might be useful if the description was more descriptive.  If I see "GET http://localhost" in the logs in a retry loop, I'm not going to know what it means.  I think "Waiting until REST API is available on %url" would be more informative.